### DISCUSSION TITLE
[Feature] Immich-cli "delete" should also delete duplicates

### DISCUSSION THREAD

**Discussion by ckocyigit** [immich-app/immich#12181] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello all,

the delete feature of the immich-cli should also delete the duplicates not only the ones which were uploaded in the current session.

Or there should be a flag/env for this behaviour.

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by febox6** [comment]
The CLI had this option, but it was removed at some point in the last year. I found a post on Reddit about it, but apparently, no one has a solution.

This option was very useful for organizing photos scattered across multiple hard drives, folders, and services. I hope it gets re-implemented back somehow.

**Reply by chefkoch** [reply]
Also was looking for that feature as well. I did not find any specific removal of the feature to delete duplicates within the commit history, but I found some rework into separate queues for  new files and  duplicates, maybe  it was a result of this one:
https://github.com/immich-app/immich/commit/eb89208abba1f1504c55a0320fd715834fdba698 @masterT:  can you give some insights on this?

I just hacked quick and dirty together an additional option for --deleteDuplicates, without actuallly updating documentation or tests. see #14769 

1. not sure if this would be an option or there are concerns in general to delete duplicates?  Some feedback from the team might be great.
2. my implemented for sure has holes and was not done correctly, but I did not do any coding for immich so far, just checked that single piece of code.

**Reply by masterT** [reply]
I encapsulated the Promise in a queue system to enable it to be retried individually instead of having to restart the whole CLI operation.

**Comment by modrin** [comment]
I second this. I uploaded all my files but forgot to use the `--delete` flag to remove the local copies. Now, when I rerun the command, it doesn’t delete the local files since they’re recognized as duplicates. I’ve tried using shell scripts to deduplicate, but the file count in Immich doesn’t match my local count, making it impossible to safely delete based on what's already uploaded. This leaves me stuck manually sorting files, which is far from ideal.

**Comment by RobinJ1995** [comment]
https://github.com/immich-app/immich/pull/20035 :)

**Comment by RobinJ1995** [comment]
fyi my changes made it into 2.2.0, there is now a `--delete-duplicates` flag that does this
